spec/withTimeoutSpec.js

```diff
@@ -0,0 +1,42 @@
+const withTimeout = require('../src/statics/withTimeout');
+const TimeoutError = require('../src/statics/TimeoutError');
+
+
+describe('withTimeout', () => {
+  it('should reject the promise with a Timeout Error if the promise does not resolve quickly', async () => {
+    const neverResolve = new Promise((resolve) => setTimeout(resolve, 10));
+    let result;
+
+    try {
+      await withTimeout(neverResolve, 1, 'The promise never resolved');
+    } catch (error) {
+      result = error;
+    }
+    expect(result.message).toEqual('The promise never resolved');
+    expect(result).toEqual(jasmine.any(TimeoutError));
+  });
+
+
+  it('should resolve if with the result of the promise if it is in time', async () => {
+    const resolves = new Promise((resolve) => {
+      setTimeout(() => { resolve('result'); }, 1);
```

Comment:
```suggestion
      setTimeout(() => resolve('result'), 1);
```

Reply:
I really don't like the unbraced syntax. If I adopted it for the full line as a rule the code gets really obfuscated in my opinion. 
```
    const fastResolve = new Promise((resolve) => setTimeout(() => resolve('result'), 1) );
```

Reply:
I really like that form lol
:man_shrugging: I guess it doesn't matter